homeassistant/components/home_connect/number.py

```diff
@@ -0,0 +1,163 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Home Connect switch."""
+
+    def get_entities():
+        """Get a list of entities."""
+        entities: HomeConnectNumberEntity = []
+        hc_api: ConfigEntryAuth = hass.data[DOMAIN][config_entry.entry_id]
+        for device_dict in hc_api.devices:
+            device = device_dict[CONF_DEVICE]
+            entities.extend(
+                HomeConnectNumberEntity(device, setting)
+                for setting in NUMBER_SETTINGS
+                if setting.key in device.appliance.status
+            )
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectNumberEntity(HomeConnectEntity, NumberEntity):
+    """Number setting class for Home Connect."""
+
+    entity_description: NumberEntityDescription
+    bsh_key: str
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        device: HomeConnectDevice,
+        entity_description: NumberEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(device, "")
+        self.bsh_key = entity_description.key
+        del self._attr_name
+        self._attr_unique_id = f"{device.appliance.haId}-{self.bsh_key}"
+        self._attr_translation_key = bsh_key_to_translation_key(self.bsh_key)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the native value of the entity."""
+        _LOGGER.debug(
+            "Tried to set value %s to %s for %s",
+            value,
+            self.bsh_key,
+            self.entity_id,
+        )
+        try:
+            await self.hass.async_add_executor_job(
+                self.device.appliance.set_setting,
+                self.bsh_key,
+                value,
+            )
+        except HomeConnectError as err:
+            _LOGGER.error(
+                "Error setting value %s to %s for %s: %s",
+                value,
+                self.bsh_key,
+                self.entity_id,
+                err,
+            )
+
+    async def async_fetch_constraints(self) -> None:
+        """Fetch the max and min values and step for the number entity."""
+        try:
+            data = await self.hass.async_add_executor_job(
+                self.device.appliance.get,
+                f"/settings/{self.bsh_key}"
+            )
+        except HomeConnectError as err:
+            _LOGGER.error("An error occurred: %s", err)
+            return
+        if not data or not (constraints := data.get(ATTR_CONSTRAINTS)):
+            return
+        self._attr_native_max_value = constraints.get(ATTR_MAX, None)
+        self._attr_native_min_value = constraints.get(ATTR_MIN, None)
+        self._attr_native_step = 1 if data.get("type", None) == "Int" else 0.1
```

Comment:
`None` is the default so you can probably remove that. Also missing `unit_of_measurement`.
```suggestion
        self._attr_native_max_value = constraints.get(ATTR_MAX)
        self._attr_native_min_value = constraints.get(ATTR_MIN)
        self._attr_native_step = constraints.get(ATTR_STEPSIZE)
        self._attr_native_unit_of_measurement = data.get(ATTR_UNIT)
```
For `native_step`, this is what I get back from the API for my fridge:

```json
{
  "data": {
    "key": "Refrigeration.FridgeFreezer.Setting.SetpointTemperatureRefrigerator",
    "constraints": {
      "min": 33,
      "max": 43,
      "stepsize": 1,
      "access": "readWrite"
    },
    "unit": "°F",
    "type": "Double",
    "value": 33
  }
}
```

Reply:
Awesome!
stepsize wasn't documented on the API docs, this is a great addition!

Reply:
Yeah, and the simulator doesn't return `stepsize` either so I don't think there would be any way to know.